mungegithub/mungers/publish_scripts/util.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This file includes functions shared by the each repository's publish scripts.
+
+# sync_repo() cherry picks the latest changes in k8s.io/kubernetes/${filter} to the
+# local copy of the repository to be published.
+#
+# overall flow
+# 1. fetch the current level of k8s.io/kubernetes
+# 2. check out the k8s.io/kubernetes HEAD into a separate branch
+# 3. rewrite the history on that branch to *only* include staging/src/k8s.io/apimachinery
+# 4. locate all commits between the last time we sync'ed and now
+# 5. switch back to the starting branch
+# 6. for each commit, cherry-pick it (which will keep authorship) into current branch
+# 7. update metadata files indicating which commits we've sync'ed to
+#
+# The function assumes to be called at the root of the repository that's going to be published.
+sync_repo() {
+    # filter could be staging/src/k8s.io/apimachinery
```

Comment:
it's a path? Better: a pattern compatible with `git filter-branch --subdirectory-filter`